lib/animated-routing.mixin.ts

```diff
@@ -0,0 +1,59 @@
+import {type LitElement} from 'lit';
+import type RouteTreeNode from './route-tree-node.js';
+import {type Context} from './page.js';
+import {RoutingMixin, type Constructor, type RoutingMixinInterface} from './routing.mixin.js';
+
+// type Constructor<T = {}> = new (...args: any[]) => T;
+
+// export declare class RoutingMixinInterface {
+//   routeEnter(currentNode: RouteTreeNode, nextNodeIfExists: RouteTreeNode | undefined, routeId: string, context: Context): Promise<boolean | void>;
+//   routeExit(currentNode: RouteTreeNode, nextNode: RouteTreeNode | undefined, routeId: string, context: Context): Promise<boolean | void>;
+// }
+
+function animatedRoutingMixin<T extends Constructor<LitElement>>(Superclass:T, className:string) {
```

Comment:
Should this be specific to `LitElement`? 
```suggestion
function animatedRoutingMixin<T extends Constructor<HTMLElement>>(Superclass:T, className:string) {
```

Reply:
i commented below but without the LitElement as the extended class, it causes typescript errors when using the lifecycle methods. @barronhagerman @jrobinson01